va/dns.go

```diff
@@ -3,37 +3,87 @@ package va
 import (
 	"context"
 	"crypto/sha256"
-	"crypto/subtle"
 	"encoding/base32"
 	"encoding/base64"
 	"errors"
 	"fmt"
 	"net/netip"
+	"slices"
 	"strings"
+	"sync"
+
+	"github.com/miekg/dns"
 
 	"github.com/letsencrypt/boulder/bdns"
 	"github.com/letsencrypt/boulder/core"
 	berrors "github.com/letsencrypt/boulder/errors"
+	"github.com/letsencrypt/boulder/iana"
 	"github.com/letsencrypt/boulder/identifier"
 )
 
 // getAddr will query for all A/AAAA records associated with hostname and return
-// the preferred address, the first netip.Addr in the addrs slice, and all
-// addresses resolved. This is the same choice made by the Go internal
+// all valid addresses resolved. This is the same choice made by the Go internal
 // resolution library used by net/http. If there is an error resolving the
 // hostname, or if no usable IP addresses are available then a berrors.DNSError
 // instance is returned with a nil netip.Addr slice.
-func (va ValidationAuthorityImpl) getAddrs(ctx context.Context, hostname string) ([]netip.Addr, bdns.ResolverAddrs, error) {
-	addrs, resolvers, err := va.dnsClient.LookupHost(ctx, hostname)
-	if err != nil {
-		return nil, resolvers, berrors.DNSError("%v", err)
+func (va ValidationAuthorityImpl) getAddrs(ctx context.Context, hostname string) ([]netip.Addr, []string, error) {
```

Comment:
Now that the middle return value doesn't have a self-documenting type, we need to document the return values in the comment.